src/common/mfu_daos.c

```diff
@@ -3580,22 +3656,26 @@ int daos_cont_serialize_hdlr(int rank, struct hdf5_args *hdf5, char *output_dir,
         oid.lo = mfu_flist_file_get_oid_low(flist, i);
         (*hdf5->oid)[i].oid_hi = oid.hi;
         (*hdf5->oid)[i].oid_low = oid.lo;
+
         rc = daos_obj_open(da->src_coh, oid, 0, &oh, NULL);
         if (rc != 0) {
             MFU_LOG(MFU_LOG_ERR, "failed to open object: "DF_RC, DP_RC(rc));
             goto out;
         }
+        ofeat = (oid.hi & OID_FMT_FEAT_MASK) >> OID_FMT_FEAT_SHIFT;
         rc = serialize_dkeys(hdf5, &dk_index, &ak_index,
-                             &oh, &i, stats);
+                             &oh, &i, da, oid, ofeat, stats);
         if (rc != 0) {
             MFU_LOG(MFU_LOG_ERR, "failed to serialize keys: %d", rc);
             goto out;
         }
         /* close source and destination object */
-        rc = daos_obj_close(oh, NULL);
-        if (rc != 0) {
-            MFU_LOG(MFU_LOG_ERR, "failed to close object: "DF_RC, DP_RC(rc));
-            goto out;
+        if (!(ofeat & DAOS_OF_ARRAY)) {
+            rc = daos_obj_close(oh, NULL);
+            if (rc != 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to close object: "DF_RC, DP_RC(rc));
+                goto out;
+            }
```

Comment:
@dsikich This is only closed if it's not OF_ARRAY, but the object is being opened above for all feature types.